nomad/state/state_store.go

```diff
@@ -241,13 +241,26 @@ func (s *StateStore) UpdateNodeDrain(index uint64, nodeID string, drain bool) er
 func (s *StateStore) NodeByID(nodeID string) (*structs.Node, error) {
 	txn := s.db.Txn(false)
 
-	existing, err := txn.First("nodes", "id", nodeID)
+	existing, err := txn.Find("nodes", "id", nodeID)
 	if err != nil {
 		return nil, fmt.Errorf("node lookup failed: %v", err)
 	}
 
 	if existing != nil {
-		return existing.(*structs.Node), nil
+		// Return exact match directly
```

Comment:
So it is best if we revert all these to their original state and then we can add additional methods such as `NodeByIDPrefix`. These methods can return a memdb.ResultIterator and use memdb.Txn().Get() and use the "_prefix" operator.

We can then add HTTP endpoints that would take a prefix and would call these new methods. And then they could return back a fixed set to the client. This would then let the user refine the search if multiple are returned or if just one is returned, the cli could just call with the full ID.